File: string/unique-morse-code-words/unique-morse-code-words.ts
function uniqueMorseRepresentations(words: string[]): number {
    const uniqueCodes = new Set<string>();
    for (let word of words) {
        const code = getCode(word);
        if (!uniqueCodes.has(code)) {
            uniqueCodes.add(code);
        }
    }

    return uniqueCodes.size;
};


const DIC =[".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."];
const A_CODE = "a".charCodeAt(0);
function getCode(word: string) : string {
    const ans : string[] = [];

    for (let c of word) {
        const code = c.charCodeAt(0) - A_CODE;
        ans.push(DIC[code]);
    }

    return ans.join("");
}

File: string/unique-morse-code-words/unique-morse-code-words.go
package uniquemorsecodewords

import "strings"

func uniqueMorseRepresentations(words []string) int {
	unique := make(map[string]bool)

	for _, word := range words {
		c := getCode(word)
		if _, ok := unique[c]; !ok {
			unique[c] = true
		}
	}

	return len(unique)
}

var code = []string{".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."}

func getCode(word string) string {
	a := rune('a')
	var ans []string
	for _, c := range word {
		ans = append(ans, code[c-a])
	}

	return strings.Join(ans, "")
}
